### PR TITLE
v2 refactor and API changes

### LINKED ISSUE #4: Request: report number of physical chips
AFAICT there's no way to know how many physical processors exist, as cpuid only reports information for the cpu the current goroutine is running on. It would be very useful (for NUMA aware programs) to have the equivelent of

```
physical id : 0
siblings    : 8
core id     : 3
cpu cores   : 4
```

from /proc/cpuinfo for all physical cpus.

### LINKED ISSUE #23: Example in Readme
Could you update output of example code in file README?
Now, I don't see L1 result.

### LINKED ISSUE #52: Change ARM implementation (SIGILL: illegal instruction)
@fwessels 

Looks like the only safe option is to use an OS call to get these features.

Maybe the Go runtime can provide some help?

https://github.com/golang/go/tree/master/src/internal/cpu

And some in the runtime:

https://github.com/golang/go/blob/8174f7fb2b64c221f7f80c9f7fd4d7eb317ac8bb/src/runtime/os_linux_arm64.go#L11

... though I haven't found where archauxv is actually called from.

### LINKED ISSUE #63: Report processors with slow AVX
A number of processors support AVX but implement it by passing the vector though a 128 bit SSE ALU twice instead of using a 256 bit ALU.  On these processors, AVX is thus not much faster than SSE.  It would be great to have a feature flag that reports whether the current processor is afflicted by this problem.  As far as I'm concerned, this affects the following processors, but I'm not super sure:

* Intel Sandy Bridge, Ivy Bridge
* AMD Jaguar/Puma
* AMD “heavy equipment” processors
* AMD Zen1

### PR DESCRIPTION
**Pull request by klauspost**
Refactors the API to make it smaller and easier to maintain.

Adds ARM64 detection.

Many tweaks and fixes.

Fixes #23
Fixes #52
Fixes #63
Fixes #4 (a bit more)
Replaces #31

### CODE REVIEW

**Comment by mythi** [comment]
nice improvements!

File: README.md [hunk 1]
```diff
@@ -15,137 +15,11 @@ Package home: https://github.com/klauspost/cpuid
...
-* **Hygon**
-
-# installing
-
-```go get github.com/klauspost/cpuid```
-
-# example
+## installing
+
+```go get github.com/klauspost/cpuid/v2```
```
Comment (mythi) [inline README.md:20]: does this require additional changes to `go.mod`  or should it be `cpuid@v2.x.y`

Comment (klauspost) [reply README.md:20]: Just changing it to the `v2.0.0` should be enough.

Comment (klauspost) [reply README.md]: I am not that much into it, but I will tweak the docs after v2 release if needed.

**Pull request merged**